### PR TITLE
Feat/7 support node 6

### PR DESCRIPTION
**Pull request by mscottx88**
Converts async/await to Promise in `/src`.

### CODE REVIEW

File: src/interceptor.ts [hunk 1]
```diff
@@ -225,6 +222,23 @@ export default class Interceptor extends EventEmitter implements IInterceptEvent
           responseSerializer,
         });
       });
+
```
Comment (mscottx88) [inline src/interceptor.ts]: I thought we should move the pipeline call until after all the event listeners are attached.

**Comment by mscottx88** [comment]
@ianwsperber I've added node 6 to travis, but because only `/src` directory was change (not `/test`) the tests will fail.

**Comment by mscottx88** [comment]
@ianwsperber i added a check to conditionally skip testing for Node 6 👍

**Comment by ianwsperber** [comment]
After seeing how this affects the build process and changes the source code I think I'm going to abandon trying to support Node 6 :/ Thanks for the work though Michael

**Pull request closed without merging**